### PR TITLE
Pausing/reloading, more enemies, customized spawn points, and more.

### PR DESCRIPTION
**Pull request by AleksandrHovhannisyan**
Game is reloaded upon player death.

Pressing P will pause/unpause the game.

Each spawn point now lets you directly modify the attributes of the enemies you spawn in (as well as their default orientation).

### CODE REVIEW

File: Assets/Scripts/Enemy.cs [hunk 1]
```diff
@@ -21,20 +21,20 @@ public class Enemy : Entity
     private float fireRateTimer = 0;
     private IWeapon fireableWeapon;
     private bool movingEnabled;
+    public bool movingEnemy = true;
     private CharacterController cc;
 
     private void Awake()
     {
+        MaxHealth = this.maxHealth;
```
Comment (klein1992) [inline Assets/Scripts/Enemy.cs]: Why move these lines here? Just curious

File: Assets/Scripts/Spawning/SpawnScript.cs [hunk 2]
```diff
@@ -21,7 +21,15 @@ public HashSet<GameObject> SpawnEnemies()
             if (enemyToSpawn != null)
             {
                 enemyToSpawn.tag = "Enemy";
-                GameObject spawnedEnemy = Instantiate(enemyToSpawn, spawnPoint.transform.position, Quaternion.identity);
+                GameObject spawnedEnemy = Instantiate(enemyToSpawn, spawnPoint.transform.position,
```
Comment (klein1992) [inline Assets/Scripts/Spawning/SpawnScript.cs]: Cool stuff, so you can specify attributes for a spawn point and an enemy will automatically "inherit" these?

File: Assets/Scripts/Enemy.cs (hunk 1 above)
Comment (AleksandrHovhannisyan) [reply Assets/Scripts/Enemy.cs]: Oh, I'm not sure to be honest. I was having a lot of trouble getting spawn points to override their enemies' stats, so I was playing around with different parts of the code trying to see what was causing the issue. Hope this didn't complicate anything for you.

File: Assets/Scripts/Spawning/SpawnScript.cs (hunk 2 above)
Comment (AleksandrHovhannisyan) [reply Assets/Scripts/Spawning/SpawnScript.cs]: Yep, each spawn point now lets you set not only the enemy to spawn but also what attributes it should come with :)

**Pull request merged**